app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -1590,6 +1594,24 @@ class StateFragmentTest {
     }
   }
 
+  @Test
+  fun testStateFragment_fractionInput_retainStateOnConfigurationChange() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      TestPlatformParameterModule.forceEnableInteractionConfigChangeStateRetention(false)
```

Comment:
Here, I am disabling Platform parameter still test is passing. Don't know why this is happening. @BenHenning can you help me to solve this issue.

Reply:
@vrajdesai78 does the new test pass when your solution is completely commented out?

Also, does it pass on Espresso when disabled?

I see one of two likely possibilities here:
1. The test isn’t verifying the thing you think it is
2. Robolectric isn’t performing the configuration change correctly 

(2) is pretty likely to be the case since we know this is an existing issue with how Robolectric interacts with Espresso’s API. I actually have a PR for introducing a new action to correctly perform rotation in Robolectric, but I ran into a lot of difficulties migrating tests over. We can look at getting at least the utility checked in so that you can use it for these new tests if the problem is indeed (2).

Reply:
@BenHenning I have tried run test in espresso and it's running infinitely, it's not the case for just one test but all tests in `StateFragmentTest.kt` are running infinitely in espresso. Also tried to run tests on develop (StateFragmentTest) and still they are running infinitely.

![Screenshot from 2022-10-22 13-07-07](https://user-images.githubusercontent.com/43074241/197330923-4ed53b93-b091-4cb9-82f4-4a4ce7d7337c.png)

I have tried to run test on develop branch (robolectric), and in develop branch also test case is passing. So, here I suspect that rotateToLandscape() is not working properly in robolectric. So now what should I try next, as espresso and robolectric both are not working.

Reply:
@vrajdesai78 I need more information here. When running on Espresso, are you running just the one test or the whole suite? (The whole suite can take almost an hour for StateFragmentTest, so I highly suggest just running the test you care about by itself). When running that test, what do you see on your emulator/device? That context is important when trying to narrow down 'running indefinitely' to a more specific issue.

Reply:
@BenHenning I am running single test on espresso and it is running infinitely. So, in emulator it is launching the screen with fraction input interaction but then it just got stuck there and nothing happens.

Reply:
Hmm I wonder if this was caused by the hint bulb anim. Can you try commenting out the hint bulb animation here: https://github.com/oppia/oppia-android/blob/b569498791d07d7d75f79c4a338cb6738dea7783/app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt#L495

Then, could you check the test again?

Reply:
Thanks Ben, after removing animation it worked.